forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/api/openbanking/event/v3_1/CallbackUrlsApiController.java

```diff
@@ -32,4 +41,19 @@
     public CallbackUrlsApiController(CallbackUrlsRepository callbackUrlsRepository, TppRepository tppRepository) {
         super(callbackUrlsRepository, tppRepository);
     }
+
+    @Override
+    protected OBCallbackUrlsResponse1 packageResponse(Collection<FRCallbackUrl1> frCallbackUrls) {
+        final List<OBCallbackUrlResponseData1> callbackUrls =
+                frCallbackUrls.stream()
+                        .filter(
+                                frCallbackUrl1 -> frCallbackUrl1.obCallbackUrl.getData().getVersion().equals(OBVersion.v3_1.getCanonicalVersion())
```

Comment:
I think it would be nice to put this in a common method - for example:

```
public static Predicate<FRCallbackUrl1> matchingVersion(OBVersion obVersion) {
   return frCallbackUrl1 -> frCallbackUrl1.obCallbackUrl.getData().getVersion().equals(obVersion.getCanonicalVersion());
}
```
Then you could just have this, which I think is a lot clearer:

```
frCallbackUrls.stream()
              .filter(matchingVersion(OBVersion.v3_1))
```

Reply:
Perhaps create a class called something like `EventsHelper` within `com.forgerock.openbanking.aspsp.rs.store.api.openbanking.event`?

Reply:
Good Suggestion!
Done!!

Reply:
Looks good! 👍